beacon-chain/db/kv/blocks.go

```diff
@@ -238,6 +238,52 @@ func (s *Store) DeleteBlock(ctx context.Context, root [32]byte) error {
 	})
 }
 
+// DeleteBlocksAndStatesBeforeSlot performs deletes all blocks and states before the given slot.
+func (s *Store) DeleteBlocksAndStatesBeforeSlot(ctx context.Context, cutoffSlot primitives.Slot) error {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.DeleteBlocksAndStatesBeforeSlot")
+	defer span.End()
+
+	// Perform all deletions in a single transaction for atomicity
+	return s.db.Update(func(tx *bolt.Tx) error {
```

Comment:
I think you are not deleting everything you should. Looking at bucket names, there are also:
- `checkpointBucket` (if this bucket stores what the name suggests, then if we remove an old checkpoint block, this bucket will still contain the checkpoint) 
- `stateValidatorsBucket` (I think this one is very important because validators are a big chunk of the state)
- `blockParentRootIndicesBucket`
- `stateSlotIndicesBucket`
- `finalizedBlockRootsIndexBucket`
- `blockRootValidatorHashesBucket`
- `originCheckpointBlockRootKey` (I don't know the implications of removing this, it's best to ask @kasey about it. The bucket is used in a few places, but after the first pruning the origin checkpoint will "move". Shouldn't we update the value in the bucket accordingly?)